File: nomad-samples/admin/js/basic-live-channel/live-channel/stop-live-channel.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";
import liveChannelStatuses from "./live-channel-statuses.js";
import waitForLiveChannelStatus from "./wait-live-channel-status.js";

/**
 * Stop a live channel
 *
 * @param {string} authToken    | Authorization token
 * @param {string} channelId    | The ID of the channel to stop
 */
export default async function stopLiveChannel(authToken, channelId) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/liveChannel/${channelId}/stop`, {
        method: "POST",
        headers: HEADERS
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {
        // Wait for the live channel to be idle
        await waitForLiveChannelStatus(authToken, channelId, liveChannelStatuses.Idle, 120, 2);
        return;
    }

    await apiExceptionHandler(RESPONSE, `Stop Live Channel ${channelId} failed`);
}

File: nomad-samples/admin/py/basic-live-channel-1.1/live_channel/stop_live_channel.py
from constants.project_constants import *
from exceptions.api_exception_handler import *
from live_channel.live_channel_statuses import *
from live_channel.wait_live_channel_status import *

def stop_live_channel(AUTH_TOKEN, CHANNEL_ID):
    # Create header for the request
    HEADERS = {
        "Content-Type": "application/json",
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    # Send the request
    try:
        RESPONSE = requests.post(ADMIN_URL + "/liveChannel/" + CHANNEL_ID + "/stop", headers= HEADERS)

        # Wait for the live channel to be idle
        wait_for_live_channel_status(AUTH_TOKEN, CHANNEL_ID, LIVE_CHANNEL_STATUSES["Idle"], 120, 2)

    except:
        api_exception_handler(RESPONSE, "Stop Live Channel " + CHANNEL_ID + " failed")

